remappers/internal/metric.go

```diff
@@ -76,3 +78,55 @@ func AddMetrics(
 		mutator(dp)
 	}
 }
+
+func Addk8sMetrics(ms pmetric.MetricSlice, mutator func(dp pmetric.NumberDataPoint), metrics ...Metric) {
+	ms.EnsureCapacity(ms.Len() + len(metrics))
+
+	for _, metric := range metrics {
+		m := ms.AppendEmpty()
+		m.SetName(metric.Name)
+
+		var dp pmetric.NumberDataPoint
+		switch metric.DataType {
+		case pmetric.MetricTypeGauge:
+			dp = m.SetEmptyGauge().DataPoints().AppendEmpty()
+		case pmetric.MetricTypeSum:
+			dp = m.SetEmptySum().DataPoints().AppendEmpty()
+		}
+
+		if metric.IntValue != nil {
+			dp.SetIntValue(*metric.IntValue)
+		} else if metric.DoubleValue != nil {
+			dp.SetDoubleValue(*metric.DoubleValue)
+		}
+
+		dp.SetTimestamp(metric.Timestamp)
+		if metric.StartTimestamp != 0 {
+			dp.SetStartTimestamp(metric.StartTimestamp)
+		}
+
+		// Calculate datastream attribute as an attribute to each datapoint
+		dataset := AddDatastream(metric.Name)
+		if dataset == "kubernetes.node" {
+			dataset = "kubernetes.pod"
+		}
+		dp.Attributes().PutStr("event.module", common.RemapperEventModule)
+		dp.Attributes().PutStr("service.type", "kubernetes")
+		dp.Attributes().PutStr(common.DatastreamDatasetLabel, dataset)
```

Comment:
For APM-Server logic we should not be setting the dataset since we want the dataset to be set via APM-Server so as to remove the dependency on any external integration for the APM-Server path. The config option `WIthSystemIntegrationDataset` for for this purpose -- when it is set to false then we won't need to deduce the dataset in the remapper.